### PR TITLE
Set UserAgent for extractor requests

### PR DESCRIPTION
**Pull request by Vampouille**

### CODE REVIEW

**Comment by pmauduit** [comment]
Looks good

**Comment by pmauduit** [comment]
... but, what about using:

```
Extractorapp (+http://www.georchestra.org/)
```

? ;-)

**Comment by fvanderbiest** [comment]
I'd say ```ExtractorApp/15.12 (geOrchestra)```

**Comment by pmauduit** [comment]
> I'd say ExtractorApp/15.12 (geOrchestra)

I'm not so fond of putting the version (like in HTTP server signatures, it provides extra unecessary information, and are often disabled anyway for security or other reasons), and keeping an URL so that people who do not know about georchestra can find info about the UA there sounds necessary to me.

**Comment by fvanderbiest** [comment]
OK for ```ExtractorApp (+http://www.georchestra.org/)```
Let's merge this ASAP.

**Comment by Vampouille** [comment]
This last commit should add URL to georchestra website on all Extractor App request

**Comment by fvanderbiest** [comment]
Please, in one word only + squash commits.

**Comment by fvanderbiest** [comment]
One last remark: would it be possible to set this UA string in a unique file instead of 3 ?

**Comment by Vampouille** [comment]
One commit, no space and one property file to set User Agent for all extractor app requests

File: extractorapp/src/main/java/org/georchestra/extractorapp/ws/extractor/WfsExtractor.java [hunk 1]
```diff
@@ -65,6 +67,8 @@
 public class WfsExtractor {
 
 	protected static final Log LOG = LogFactory.getLog(WcsExtractor.class.getPackage().getName());
+    @Autowired
```
Comment (fvanderbiest) [inline extractorapp/src/main/java/org/georchestra/extractorapp/ws/extractor/WfsExtractor.java]: indents ?

File: extractorapp/src/main/filtered-resources/WEB-INF/ws-servlet.xml [hunk 2]
```diff
@@ -9,6 +9,8 @@
 
     <context:annotation-config/>
 
+    <context:property-placeholder location="/WEB-INF/extractorApp.properties"/>
```
Comment (fvanderbiest) [inline extractorapp/src/main/filtered-resources/WEB-INF/ws-servlet.xml]: extractorapp.properties ?

**Comment by fvanderbiest** [comment]
Don't you think this PR should also modify https://github.com/georchestra/georchestra/blob/master/security-proxy/src/deb/resources/etc/georchestra/security-proxy/user-agents.properties (and a new PR made for georchestra/datadir) and https://github.com/georchestra/georchestra/blob/master/security-proxy/src/main/webapp/WEB-INF/applicationContext-security.xml#L81-L92 ?

File: extractorapp/src/main/java/org/georchestra/extractorapp/ws/extractor/csw/CSWExtractor.java [hunk 3]
```diff
@@ -32,7 +34,10 @@
 public class CSWExtractor {
 	
 	protected static final Log LOG = LogFactory.getLog(CSWExtractor.class.getPackage().getName());
-	
+
+    @Autowired
```
Comment (fvanderbiest) [inline extractorapp/src/main/java/org/georchestra/extractorapp/ws/extractor/csw/CSWExtractor.java]: indents again here

File: extractorapp/src/main/java/org/georchestra/extractorapp/ws/extractor/WfsExtractor.java [hunk 4]
```diff
@@ -64,7 +66,9 @@
  */
 public class WfsExtractor {
 
-	protected static final Log LOG = LogFactory.getLog(WcsExtractor.class.getPackage().getName());
+    protected static final Log LOG = LogFactory.getLog(WcsExtractor.class.getPackage().getName());
+    @Autowired
+    private Environment env;
```
Comment (fvanderbiest) [inline extractorapp/src/main/java/org/georchestra/extractorapp/ws/extractor/WfsExtractor.java]: indents here

**Comment by fvanderbiest** [comment]
OK, indents will be taken care of later.

**Comment by fvanderbiest** [comment]
... but the testsuite seems broken => not merging until fixed.

**Comment by Vampouille** [comment]
It seems that : https://github.com/georchestra/georchestra/blob/Set_Dedicated_UserAgent_960/extractorapp/src/main/java/org/georchestra/extractorapp/ws/extractor/WcsExtractor.java#L39 cannot be autowired on travis. 
So we get npo here : https://github.com/georchestra/georchestra/blob/Set_Dedicated_UserAgent_960/extractorapp/src/main/java/org/georchestra/extractorapp/ws/extractor/WcsExtractor.java#L53

**Comment by Vampouille** [comment]
Test suite is now working

**Comment by fvanderbiest** [comment]
This looks good, please merge.

**Pull request merged**